refinery/tool_manager/tests.py

```diff
@@ -3473,3 +3473,75 @@ def test_get_workflows_with_connection_error(self):
                 ),
                 context.exception.message
             )
+
+
+class ParameterTests(ToolManagerTestBase):
+    def test_cast_param_value_to_proper_type_bool(self):
+        parameter = ParameterFactory(
+            name="Bool Param",
+            description="Boolean Parameter",
+            value_type=Parameter.BOOLEAN,
+            default_value="False"
+        )
+        self.assertFalse(
+            parameter.cast_param_value_to_proper_type(parameter.default_value)
+        )
+
+        parameter = ParameterFactory(
+            name="Bool Param",
+            description="Boolean Parameter",
+            value_type=Parameter.BOOLEAN,
+            default_value="True"
+        )
+        self.assertTrue(
+            parameter.cast_param_value_to_proper_type(parameter.default_value)
+        )
+
+    def test_cast_param_value_to_proper_type_string(self):
+        for string_type in Parameter.STRING_TYPES:
+            parameter = ParameterFactory(
+                name="String Param",
+                description="String Parameter",
+                value_type=string_type,
+                default_value="Coffee"
+            )
+            self.assertEqual(
+                str,
+                type(
+                    parameter.cast_param_value_to_proper_type(
+                        parameter.default_value
+                    )
+                )
+            )
+
+    def test_cast_param_value_to_proper_type_int(self):
+        parameter = ParameterFactory(
+            name="Int Param",
+            description="Integer Parameter",
+            value_type=Parameter.INTEGER,
+            default_value="1"
+        )
+        self.assertEqual(
+            int,
+            type(
+                parameter.cast_param_value_to_proper_type(
+                    parameter.default_value
+                )
+            )
+        )
+
+    def test_cast_param_value_to_proper_type_float(self):
+        parameter = ParameterFactory(
+            name="Float Param",
+            description="Float Parameter",
+            value_type=Parameter.FLOAT,
+            default_value="1.0"
+        )
+        self.assertEqual(
+            float,
+            type(
+                parameter.cast_param_value_to_proper_type(
+                    parameter.default_value
+                )
+            )
+        )
```

Comment:
In the tests, I think it might be better just to pass in plain values, rather than pulling `parameter.default_value`: Imagine a scenario where default_value starts returning something that's already been cast to the right type, and then at some later point cast_param stops casting: These tests might not catch either of those changes.

Does the ParameterFactory require all those fields, or could you make it more concise for the sake of readablity? Nothing by `value_type` really matters here...

And maybe just do the assertEqual on the value, instead of the type of the value? Shorter, and stronger. (Though I get that type is the thing you're really interested in here.)